### ISSUE TITLE
glfwUpdateGamepadMappings does not work

### ISSUE THREAD

**Issue by DesertCookie** [glfw/glfw#1422]
I'm trying to change my controller's mapping because several buttons aren't assigned correctly.
I tried modifying the default GLFW mapping for XInput controllers (Xbox 360 Controller F310 by Logitec in my case) and applying it by using `glfwUpdateGamepadMappings(String)`.
My mapping is: `78696e70757401000000000000000000,XInput Gamepad (GLFW),platform:Windows,a:b0,
b:b1,x:b2,y:b3,leftshoulder:b4,rightshoulder:b5,back:b6,start:b7,leftstick:b8,
rightstick:b9,leftx:a0,lefty:a1~,rightx:a2,righty:a3,lefttrigger:a4,
righttrigger:a5,dpup:h0.1,dpright:h0.2,dpdown:h0.4,dpleft:h0.8` (for now I'm only trying to invert the left stick's y-axes).

I have verified that the specified name and GUID are the same as my controller; no changes made in the mapping string seem to be applied, my controller's behaviour doesn't change.

Keys that are assigned wrongly include:
- start instead of leftstick
- rightstick instead of leftstick
- dpadup instead of leftstick
- ...

I'm using LWJGL 3.2.1 with GLFW 3.3.0.

**Comment by elmindreda** [comment]
Did GLFW emit any error when you called `glfwUpdateGamepadMappings`?

**Comment by DesertCookie** [comment]
> Did GLFW emit any error when you called `glfwUpdateGamepadMappings`?
No, no output whatsoever.

**Comment by elmindreda** [comment]
The mapping you provided seems to parse correctly. The only difference I can see between it and the built-in mapping is inverting the left Y-axis, which seems to work correctly on this setup.

**Comment by elmindreda** [comment]
Tested with GLFW master on Windows 10 and a wired 360 controller.

**Comment by EquilinoxModKit** [comment]
Since I use LWJGL, could it be a problem with it then?

**Comment by DesertCookie** [comment]
Since I'm using LWJGL, is it likely is it likely that the problem lies with it?

**Comment by elmindreda** [comment]
That is unlikely.

It's possible that your copy of GLFW is older than the addition of support for `~`.

**Comment by DesertCookie** [comment]
Could you maybe exist me with how to change button mappings? I tried just swapping `b9` for the right stick and `h.01` for dpad up but that doesn't work. I don't know much about the letters and didn't have time to read into the docs about them. Should my swap even work?

**Comment by elmindreda** [comment]
Yes, it should work to swap any button, hat direction or axis for any other. It sounds like the mapping isn't being applied at all, which is strange.

Is the mapping string you pass in all one line with no extra whitespace?

**Comment by DesertCookie** [comment]
It is one continuous string.

**Comment by JasperNV** [comment]
This is because the XInput controller is created at platform init time, and glfwUpdateGamepadMappings doesn't update mappings for existing controllers.

**Comment by DesertCookie** [comment]
GLFW init time or when a controller gets connected?

**Comment by elmindreda** [comment]
@JasperNV It should be doing that [here](https://github.com/glfw/glfw/blob/90e22947c60b0c1fa47cf1496790ce1942e4a5d8/src/input.c#L1136).
